.github/workflows/pkg-js-release.yaml

```diff
@@ -0,0 +1,67 @@
+name: Release (JS)
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - "pkg/js/v*"
+
+jobs:
+  test:
+    uses: ./.github/workflows/pkg-js-build.yaml
+    secrets: inherit
+
+  publish:
+    needs: [test]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    defaults:
+      run:
+       working-directory: "pkg/js"
+       
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        with:
+          fetch-depth: 0
+          path: "."
+          sparse-checkout: "pkg/js"
+          sparse-checkout-cone-mode: false
+
+      - name: Set up node
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          node-version: 20
+          registry-url: "https://registry.npmjs.org"
+          scope: "@openfga"
+          always-auth: false
+          cache: "npm"
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Publish to npm
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  release:
+    runs-on: ubuntu-latest
+    needs: [publish]
+    permissions:
+      contents: write
+      packages: write # publish a new github release
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        with:
+          fetch-depth: 0
+
+      - uses: Roang-zero1/github-create-release-action@57eb9bdce7a964e48788b9e78b5ac766cb684803 # v3.0.1
+        with:
+          version_regex: ^pkg/js/v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+
+          prerelease_regex: ^pkg/js/v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+-(alpha|beta)$
```

Comment:
Can you add support for `.d+` after alpha beta?

So v0.1.0-beta.0 or v0.1.0-beta.21, not v0.1.0-beta? (would be cool to add this restriction to line 7 of this file too)

Reply:
All releases or just js?

Reply:
Line 7 is proving difficult as its a pattern match, not a regex.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Will something like `pkg/js/v[0-9]+.[0-9]+.[0-9]+*` work?

It should match `pkg/js/v0.2.0`, as well as `pkg/js/v0.2.0-beta`

Reply:
Let's keep line 7 the same then, but abort early down the line.

Note: pkg/js/v0.2.0-beta should not be accepted - pkg/js/v0.2.0-beta.0 should be